### PR TITLE
Harden NettyTransportSpec

### LINKED ISSUE #25479: failed: NettyTransportSpec
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/6282/

```
[INFO] [08/16/2018 04:52:25.545] [pool-1-thread-1-ScalaTest-running-NettyTransportSpec] [akka.remote.Remoting] Starting remoting
[INFO] [08/16/2018 04:52:25.552] [pool-1-thread-1-ScalaTest-running-NettyTransportSpec] [akka.remote.Remoting] Remoting started; listening on addresses :[akka.tcp://sys@127.0.1.1:41755]
[INFO] [08/16/2018 04:52:25.552] [pool-1-thread-1-ScalaTest-running-NettyTransportSpec] [akka.remote.Remoting] Remoting now listens on addresses: [akka.tcp://sys@127.0.1.1:41755]
[info] - should bind to a random port but remoting accepts from a specified port *** FAILED *** (21 milliseconds)
[info]   List(tcp://sys@127.0.1.1:41755) contained element tcp://sys@127.0.1.1:41755 (NettyTransportSpec.scala:72)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.MatchersHelper$.indicateFailure(MatchersHelper.scala:346)
[info]   at org.scalatest.words.ResultOfNotWordForAny.contain(ResultOfNotWordForAny.scala:475)
[info]   at akka.remote.transport.netty.NettyTransportSpec$$anonfun$1$$anonfun$apply$mcV$sp$2.apply(NettyTransportSpec.scala:72)
[info]   at akka.remote.transport.netty.NettyTransportSpec$$anonfun$1$$anonfun$apply$mcV$sp$2.apply(NettyTransportSpec.scala:60)
```

### PR DESCRIPTION
**Pull request by chbatey**
Keep the temporary port open while the ActorSystem is created so it
can't select the same port to bind on.

Fixes #25479

### CODE REVIEW

File: akka-remote/src/test/scala/akka/remote/transport/netty/NettyTransportSpec.scala [hunk 1]
```diff
@@ -58,26 +62,39 @@ class NettyTransportSpec extends WordSpec with Matchers with BindBehavior {
     }
 
     "bind to a random port but remoting accepts from a specified port" in {
-      val address = SocketUtil.temporaryServerAddress(InetAddress.getLocalHost.getHostAddress, udp = false)
+      val (openSS, address) = randomOpenServerSocket()
```
Comment (patriknw) [inline akka-remote/src/test/scala/akka/remote/transport/netty/NettyTransportSpec.scala]: perhaps add a comment why this socket is kept open

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by raboof** [comment]
PLS BUILD

**Review by jrudolph — APPROVED** [review]
Change LGTM, though, I don't understand that feature at all ;). Why would it be useful to bind to a random port? Also the test doesn't seem to assert anything particularly useful aside from that "random port doesn't try to use already open port" but maybe I just don't understand the feature...

**Comment by jrudolph** [comment]
Sounds to me as if the children's room needs some tidying up if you need a robot to find a cave in a corner where to put your server rack. Also the robot will then put down a cable between it and ~your Real Data Center~ ~the server room~ the kitchen (obviously). The cable will mess up the children's racing track and also one car was repurposed as the side-car to drive the end of the cable to the kitchen. The stage is set. First act. In the corridor. The cat just came home from "work"

File: akka-testkit/src/main/scala/akka/testkit/SocketUtil.scala [hunk 2]
```diff
@@ -24,7 +24,7 @@ object SocketUtil {
       SocketUtil.temporaryServerAddress(address = "127.20.0.0")
       true
     } catch {
-      case e: java.net.BindException ⇒
+      case _: java.net.BindException ⇒
         false
     }
   }
```
Comment (helena) [inline akka-testkit/src/main/scala/akka/testkit/SocketUtil.scala]: Out of scope for your change, but seems cleaner
```suggestion
private val canBindOnAlternativeLoopbackAddresses: Boolean =
  Try(SocketUtil.temporaryServerAddress(address = "127.20.0.0"))
    .recover { case _: java.net.BindException ⇒ false}.isSuccess
```

**Pull request merged**